src/main/java/org/mockito/internal/junit/JUnitTool.java

```diff
@@ -4,15 +4,32 @@
  */
 package org.mockito.internal.junit;
 
+import org.mockito.exceptions.verification.ArgumentsAreDifferent;
+
 public class JUnitTool {
 
-    private static final JUnitDetecter detecter = new JUnitDetecter();
+    private final static boolean hasJUnit = canCreateJUnitClass();
+
+    private JUnitTool() {
+    }
 
-    public static boolean hasJUnit() {
-        return detecter.hasJUnit();
+    /**     
+     * If JUnit is used,  AssertionError is returned that extends from ComparisonFailure      
+     * and hence provide a better IDE support as the comparison result is comparable      
+     */
+    public static AssertionError createArgumentsAreDifferentException(String message, String wanted, String actual) {
+        if (hasJUnit) {
+            return new org.mockito.exceptions.verification.junit.ArgumentsAreDifferent(message, wanted, actual);
```

Comment:
Can you test if this actually work in a project without JUnit? I want to make sure that the code does not throw class not found error or something like that. The previous code used reflection to construct the instance of the class that required JUnit on the classpath.

Perhaps we already validate this condition in our project (testng submodule) but I'm not sure right now. I will double check that over the weekend.